src/smtml/expr.ml

```diff
@@ -420,11 +425,16 @@ let rec relop ty op hte1 hte2 =
   | Ne, _, Val (App (`Op "symbol", [ Str _ ]))
   | Ne, Val (App (`Op "symbol", [ Str _ ])), _ ->
     value True
+  | Eq, (Symbol s1 as e1), (Symbol s2 as e2) ->
+    if Symbol.equal s1 s2 then
+      make (Unop (Ty_bool, Not, make (Unop (Ty_bool, Is_nan, make e1))))
```

Comment:
Related to the previous comment in `Unop (ty, _, _)`, is `ty` the resulting type or the type parameter, because if it is the type parameter then  `Unop (Ty_bool, Is_nan, make e1)` is incorrect, type of `e1` should be used instead of `Ty_bool`.
Since the type is not an option, and there are operations which are not polymorphic (like `Not`) I assume that it is the type of the resulting term from the application.